manta-accounting/src/transfer/sender.rs

```diff
@@ -109,6 +109,27 @@ where
         Self::new(secret, utxo, nullifier)
     }
 
+    /// Samples a new [`PreSender`], returning also its [`Utxo`].
+    ///
+    /// [`Utxo`]: crate::transfer::utxo::UtxoType::Utxo
+    #[inline]
+    pub fn sample_with_utxo<R>(
+        parameters: &S,
+        authorization_context: &mut S::AuthorizationContext,
+        identifier: S::Identifier,
+        asset: S::Asset,
+        rng: &mut R,
```

Comment:
Doesn't randomness need to be generated outside of library

I think right now signer would generate the randomness not a verification server

Could be mistaken